app/templates/components/user-menu.hbs

```diff
@@ -0,0 +1,33 @@
+{{#if session.isOpening}}
+  <li class="dropdown disabled signing-in">Signing in...</li>
+{{else if session.isAuthenticated}}
+  <span class="user-avatar"><img src="{{session.currentUser.avatarUrl32}}" /></span>
+  <li class="dropdown">
+     <a class="dropdown-toggle" data-toggle="dropdown" href="#">
+       {{session.currentUser.login}} <b class="caret"></b>
+     </a>
+     <ul class="dropdown-menu dropdown-menu-right">
+       <li><a {{action 'signOut'}} class="test-sign-out">Sign out</a></li>
+     </ul>
+  </li>
+{{else}}
```

Comment:
Seems like the `else if session.isAuthenticated` and `else` block are roughly the same, I'd love to remove this duplication.  Looks like the following might do it:

* Add an `avatarURL` and `userName` computeds to the component that defaults to `null` if `session.isAuthenticated` is false.
* Use `{{action (if session.isAuthenticated "signOut" "signInViaGithub")}}` to get the right action.